hamster_gtk/tracking/screens.py

```diff
@@ -230,6 +239,27 @@ def reset(self):
         """Clear all data entry fields."""
         self.raw_fact_entry.props.text = ''
 
+    def set_raw_fact(self, raw_fact):
+        """Set the text in the raw fact entry."""
+        self.raw_fact_entry.props.text = raw_fact
+
+    def _get_recent_activities_widget(self):
+        scrolled_window = Gtk.ScrolledWindow()
+        scrolled_window.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
+        grid = RecentActivitiesGrid(self, self._app.controller)
+        # We need to 'show' the grid early in order to make sure space is
+        # allocated to its children so they actually have a height that we can
+        # use.
+        grid.show_all()
+        # We fetch an arbitrary Button as height-reference
```

Comment:
this is really ugly and crashes when there are no items:

```
$ env XDG_CONFIG_HOME=$HOME/.dev_config XDG_DATA_HOME=$HOME/.dev_local hamster-gtk
Hamster-GTK started.
Traceback (most recent call last):
  File "/nix/store/fzrxvdjas3icqb2n5akahvix758jnfvj-python3.6-hamster-gtk/lib/python3.6/site-packages/hamster_gtk/hamster_gtk.py", line 213, in _activate
    self.window = MainWindow(app)
  File "/nix/store/fzrxvdjas3icqb2n5akahvix758jnfvj-python3.6-hamster-gtk/lib/python3.6/site-packages/hamster_gtk/hamster_gtk.py", line 107, in __init__
    self.add(TrackingScreen(self.app))
  File "/nix/store/fzrxvdjas3icqb2n5akahvix758jnfvj-python3.6-hamster-gtk/lib/python3.6/site-packages/hamster_gtk/tracking/screens.py", line 49, in __init__
    self.start_tracking_view = StartTrackingBox(self._app)
  File "/nix/store/fzrxvdjas3icqb2n5akahvix758jnfvj-python3.6-hamster-gtk/lib/python3.6/site-packages/hamster_gtk/tracking/screens.py", line 195, in __init__
    self.recent_activities_widget = self._get_recent_activities_widget()
  File "/nix/store/fzrxvdjas3icqb2n5akahvix758jnfvj-python3.6-hamster-gtk/lib/python3.6/site-packages/hamster_gtk/tracking/screens.py", line 255, in _get_recent_activities_widget
    child = grid.get_children()[1]
IndexError: list index out of range
Traceback (most recent call last):
  File "/nix/store/fzrxvdjas3icqb2n5akahvix758jnfvj-python3.6-hamster-gtk/lib/python3.6/site-packages/hamster_gtk/hamster_gtk.py", line 218, in _activate
    app.add_window(self.window)
TypeError: Argument 1 does not allow None as a value
Hamster-GTK shut down.
```

Reply:
I agree that the way we derive our min_height hint is far from elegant. I am absolutly open to better approaches! Empty children are definitely a nice catch and have been addressed in the latest version.

Reply:
Maybe add a todo comment and go with this.

Reply:
I created #224  as I would like to avoid todo comments.